korlibs-concurrent/src/korlibs/concurrent/lock/Lock.kt

```diff
@@ -0,0 +1,64 @@
+@file:Suppress("PackageDirectoryMismatch")
+
+package korlibs.concurrent.lock
+
+import korlibs.concurrent.thread.*
+import kotlin.time.*
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+
+interface BaseLock {
+    fun notify(unit: Unit = Unit)
+    fun wait(time: Duration): Boolean
+    //fun lock()
+    //fun unlock()
+}
+
+//typealias Lock = BaseLock
```

Comment:
We should not leave in comment code in PR. YAGNI: You aren't gonna need it. If you do, you can readd it from earlier commit.

Reply:
This is old code that was like that, I'm okay with removing those lines. So feel free to do so.